File: app/src/main/java/com/annapol04/munchkin/db/EventDataTypeConverter.java
package com.annapol04.munchkin.db;

import android.arch.persistence.room.TypeConverter;

import com.annapol04.munchkin.engine.Decoder;
import com.annapol04.munchkin.engine.EventData;

public class EventDataTypeConverter {
    private static final Decoder decoder = new Decoder();

    @TypeConverter
    public static byte[] toBytes(EventData data) {
        return data.getBytes();
    }

    @TypeConverter
    public static EventData toEventData(byte[] data) {
        return decoder.decodeEventData(data);
    }
}

File: app/src/main/java/com/annapol04/munchkin/db/EventDataTypeConverter.kt
package com.annapol04.munchkin.db

import android.arch.persistence.room.TypeConverter

import com.annapol04.munchkin.engine.Decoder
import com.annapol04.munchkin.engine.EventData

class EventDataTypeConverter {
    private val decoder = Decoder()

    @TypeConverter
    fun toBytes(data: EventData): ByteArray {
        return data.getBytes()
    }

    @TypeConverter
    fun toEventData(data: ByteArray): EventData? {
        return decoder.decodeEventData(data)
    }
}
